src/App.tsx

```diff
@@ -3,23 +3,29 @@ import { BrowserRouter, Routes, Route } from "react-router-dom";
 import Login from "./login/pages/Login";
 import PageNotFound from "./components/PageNotFound";
 
-import Layout from "./components/Layout";
-import ProductPage from "./products/pages/ProductPage";
+import DashboardLayout from "./components/DashboardLayout";
+import ProductsPage from "./products/pages/ProductsPage";
 
 import "./App.css";
 import CartPage from "./cart/pages/CartPage";
+import MainLayout from "./components/MainLayout";
+import Product from "./products/pages/Product";
+// import Protected from "./components/Protected";
 
 function App() {
   return (
     <>
       <BrowserRouter>
         <Routes>
-          <Route path="/" element={<Layout />}>
-            <Route path="/" element={<ProductPage />} />
-
+          <Route path="/" element={<MainLayout />}>
             <Route path="/login" element={<Login />} />
-            <Route path="/cart" element={<CartPage />} />
-            <Route path="*" element={<PageNotFound />} />
+
+            <Route path="/" element={<DashboardLayout />}>
```

Comment:
it's better to place routes in a constant file.

const uiRotues = {
    cart: '/cart',
    home: '/',
    ....
}